### PR TITLE
Week3

### PR DESCRIPTION
**Pull request by stephskas**
Adding express lessons 1-7 in answers directory, adding practiceExpress.js and new-public directory with index.html

### CODE REVIEW

File: 02-express-tutorial/answers/06-alternative-flavor.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+module.exports.items = ['item1', 'item2']
+const person = {
+  name: 'bob',
+}
+
+module.exports.singlePerson = person
```
Comment (akosasante) [inline 02-express-tutorial/answers/06-alternative-flavor.js]: I think this file got accidentally carried over from the Week 1 content. In the Express week, you should instead have a file here called `06-basic-json.js`

File: 02-express-tutorial/answers/practice1.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+		writeFile('./content/practice.txt', `${sentence}\n${username}`, (err) => {
+			if (err) throw err;
+			console.log('file created');
+		});
+	} else {
+		console.log('sentence is undefined');
+	}
+} catch (err) {
+	console.log(err);
+}
```
Comment (akosasante) [inline 02-express-tutorial/answers/practice1.js:17]: I think this along with `02-express-tutorial/answers/practice2.js` and `02-express-tutorial/answers/practice3.js` may have accidentally gotten copied over from the `01-node-tutorial` directory

File: 02-express-tutorial/final/06-basic-json.js [hunk 3]
```diff
@@ -1,10 +1,15 @@
...
+app.get('*', (req, res) => {
+	res.status(404).send('resource not found');
+});
 
-app.listen(5000, () => {
-  console.log('Server is listening on port 5000....')
-})
+app.listen(5002, () => {
+	console.log('server is listening on port: 5002...');
+});
```
Comment (akosasante) [inline 02-express-tutorial/final/06-basic-json.js:15]: I think these changes should have been made in the `answers` directory instead

File: 02-express-tutorial/final/07-params-query.js [hunk 4]
```diff
@@ -1,58 +1,61 @@
-const express = require('express')
-const app = express()
-const { products } = require('./data')
+const express = require('express');
```
Comment (akosasante) [inline 02-express-tutorial/final/07-params-query.js:1]: I think the changes in this file should have been made in the `answers` directory instead

**Review by akosasante — COMMENTED** [review]
For the Week 3 homework, modules 01 to 07 in the `answers` directory were reviewed and look good. The only thing missing in the `answers` folder is `06-basic-json.js`.

There seems to be some changes that were made outside of the answers folder, and a lot of changes carried over from the Week 1 and Week 2 PRs. For future reference, I would recommend, that whenever you start working on a new week's assignment (even if you're just working ahead/early and the previous assignments haven't been reviewed yet), it's usually easier to make your branch off of main. That means following these steps:

- Make sure you have no changes currently in the repo. If you do you can temporarily hide them by doing: `git stash save`
- switch to the main branch in your local repo: `git checkout main`
- pull any changes from the Github remote fork: `git pull origin`
- make a new branch for the assignment you're currently working on: `git checkout -b week3`
- Make your changes!. If you wanted to continue working on the changes you temporarily hid in step one, you can bring them back with: `git stash pop`

Another tip I would give, is to [commit early and often](https://deepsource.io/blog/git-best-practices/#3-commit-early-commit-often)! So if you're working on file `01-http-basics.js` and you finish working on it and are about to move on to `02-http-app.js`, this is a good point to stop, and commit your work so far:

- Add just the files you want to include in this commit: `git add 02-express-tutorial/answers/01-http-basics.js`
-  Make the commit, with a clear message of what it's about: `git commit -m "Worked on HTTP Basics"`

It's totally normal to have lots of small commits in a PR and it can make it easier for others to review your changes. As well, avoiding using `git add .` and instead adding the specific files or folders that you intend to, makes it easier for you to only commit the changes that you mean to and not include extra files accidentally. Using `git status` to see what files have changed, will also help you notice if things have been accidentally moved or copied to other directories while you were testing things.

File: 02-express-tutorial/answers/06-alternative-flavor.js (hunk 1 above)
Comment (stephskas) [reply 02-express-tutorial/answers/06-alternative-flavor.js]: Ahhh yes, I got too far ahead and now I'm not sure how to fix the pull requests for each week. I did complete the entire video and was writing to the app.js file which was outside of the answers directory, and then after I completed the lesson I moved the contents of app.js file into a new file in answers with the provided name. I didn't encounter any issues with the lessons but should have committed earlier and often as you suggested! And not worked so far ahead without confirming I was set up correctly. Thank you for the suggestions! I will follow your guidelines going forward.

Comment (akosasante) [reply 02-express-tutorial/answers/06-alternative-flavor.js]: All good; totally understand! 

Probably the most straightforward way will be to just merge the PRs one by one after resolving any feedback from the review comments, starting with week 1, then 2, 3, 4. Then if there's some cleanup with the folders or anything like that that needs to be done still, make a separate cleanup branch and PR for that, branched off of `main` _after_ you've merged all the PRs and done `git pull main` locally.

**Pull request merged**